app/components/formatted_markdown_text.js

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {Text} from 'react-native';
+import PropTypes from 'prop-types';
+import Renderer from 'commonmark-react-renderer';
+import {Parser} from 'commonmark';
+import {injectIntl, intlShape} from 'react-intl';
+import MarkdownLink from 'app/components/markdown/markdown_link';
+import {concatStyles, changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import {getMarkdownTextStyles} from 'app/utils/markdown';
+
+const TARGET_BLANK_URL_PREFIX = '!';
+
+/*
+* Translations component with a similar API to <FormattedText> component except the message string
+* accepts markdown. It supports the following non-block-level markdown:
+* - *italic*
+* - **bold**
+* - `inline code`
+* - ~~strikethrough~~
+* - [link](http://example.com/)
+* - line\nbreaks
+*
+* Note: Line breaks (\n) in a defaultMessage parameter string must be surrounded by curly brackets {} in JSX. Example:
+* <FormattedMarkdownText id='my.example' defaultMessage={'first line\nsecond line'} theme={theme} />
+*/
+class FormattedMarkdownText extends React.PureComponent {
+    static propTypes = {
+        intl: intlShape.isRequired,
+        id: PropTypes.string.isRequired,
+        theme: PropTypes.object.isRequired,
+        defaultMessage: PropTypes.string,
+        values: PropTypes.object,
+        style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
+    };
+
+    static defaultProps = {
+        defaultMessage: '',
+    };
+
+    constructor(props) {
+        super(props);
+        this.parser = this.createParser();
+        this.renderer = this.createRenderer();
+        this.textStyles = getMarkdownTextStyles(props.theme);
+        this.baseTextStyle = getStyleSheet(props.theme).message;
+    }
+
+    createParser = () => {
+        return new Parser();
+    }
+
+    createRenderer = () => {
+        return new Renderer({
+            renderers: {
+                text: this.renderText,
+                emph: Renderer.forwardChildren,
+                strong: Renderer.forwardChildren,
+                code: this.renderCodeSpan,
+                link: this.renderLink,
+                hardBreak: this.renderBreak,
+                softBreak: this.renderBreak,
+                paragraph: this.renderParagraph,
+                del: Renderer.forwardChildren,
+            },
+        });
+    }
+
+    computeTextStyle = (baseStyle, context) => {
+        return concatStyles(baseStyle, context.map((type) => this.textStyles[type]));
+    }
+
+    renderText = ({context, literal}) => {
+        const style = this.computeTextStyle(this.props.style || this.baseTextStyle, context);
```

Comment:
Does this seem correct to take the style property (if present) *or* the baseTextStyle? Alternatively should they be combined?

Reply:
I think having `this.props.style` replace the base style like this makes sense. If you're going through the effort of styling it, you can make sure it's using the theme and everything correctly